compiler-core/src/analyse.rs

```diff
@@ -1762,10 +1762,10 @@ fn get_compatible_record_fields<A: std::fmt::Debug>(
             };
 
             // The labels must be the same
-            if !argument
+            if argument
                 .label
                 .as_ref()
-                .is_some_and(|(_, arg_label)| arg_label == first_label)
+                .is_none_or(|(_, arg_label)| arg_label != first_label)
```

Comment:
this is so weird

Reply:
I thought so too. We could disable this lint if we wanted

Reply:
If it comes up a few times let's give it a think